### ISSUE TITLE
E2Term Failure 

### ISSUE THREAD

**Issue by MKSBarbosa** [wineslab/o-ran-e2sim#5]
Hello, everyone.

I attempted to test the scenario-zero.cc following the tutorial [[https://openrangym.com/tutorials/ns-o-ran](https://openrangym.com/tutorials/ns-o-ran)], but it seems there is an issue with ASN.1. According to the ns3 log, the gNB never receives a response from e2term after sending the setup request. 

![image](https://github.com/user-attachments/assets/02720023-5467-4e2b-a439-de32fb72c48f)

This issue is also evident in the pcap capture. 

![image](https://github.com/user-attachments/assets/06cc8a93-fee3-4ea1-aee3-9e552bf462db)

Additionally, the e2term log does not display the expected output, as shown in the provided example.

![image](https://github.com/user-attachments/assets/b35d9e67-0859-4e38-a755-b4c68325f734)

Could someone please assist me with this problem? I am using Ubuntu 20.04.

Thank you!
[e2term.log](https://github.com/user-attachments/files/16563958/e2term.log)
[ns3-scenario-zero.log](https://github.com/user-attachments/files/16563960/ns3-scenario-zero.log)

**Comment by ethanlin355** [comment]
I encounter this issue too,I'm using F-release of the near rt ric,I think the issue is there are 2 protocol IE id is missing 49 and 50 inside of the asn1c folder but i try to rebuild the program the issue still exsist

**Comment by ethanlin355** [comment]
9.3.7 Constant definitions

 

id-GlobalE2node-ID ProtocolIE-ID ::= 3

id-RANfunctionsAdded ProtocolIE-ID ::= 10

id-TransactionID ProtocolIE-ID ::= 49

id-E2nodeComponentConfigAddition ProtocolIE-ID ::= 50

GlobalE2node-ID:

Base station's PLMN ID and GNB ID

RANfunctionsAdded:

Base station supported RAN Functions and parameters， including RANfunctionID, RANfunctionOID 

xApp will choose accroding to the list subscription's RANfunctionID

TransactionID:

E2 Node generate identifyiers, RIC reply RESPONSE will fill in TransactionID

E2nodeComponentConfigAddition:

Component configuration of the base station, such as F1, S1, X2 and other parameters.

**Comment by Thecave3** [comment]
Hi all, 

Can you try using the configuration provided here: [https://github.com/o-ran-sc/ric-plt-ric-dep/blob/master/RECIPE_EXAMPLE/example_recipe_oran_cherry_release.yaml](https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fo-ran-sc%2Fric-plt-ric-dep%2Fblob%2Fmaster%2FRECIPE_EXAMPLE%2Fexample_recipe_oran_cherry_release.yaml&data=05%7C02%7Clacava.a%40northeastern.edu%7C61f23efce1654b2212dc08dd3fbf6880%7Ca8eec281aaa34daeac9b9a398b9215e7%7C0%7C0%7C638736813557336133%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=DI7YnsadTlTBM6xiRfvfv8s0oVcSFK0lqU5PapTljWE%3D&reserved=0) 

for the dbaas, e2term, rtmgr and submgr except from the e2mgr where you should use the 5.4.15 version (the 5.4.16 was not available)?

**Comment by ethanlin355** [comment]
I installed the cherry release on my smo but still i can't find any successful connection evidence in wireshark

![Image](https://github.com/user-attachments/assets/c0234ef9-8ee8-4ce2-8ac4-15435a427af4)
![Image](https://github.com/user-attachments/assets/aee984ad-5f9b-4e1e-ae8e-bc61f997004b)
![Image](https://github.com/user-attachments/assets/419f1890-348c-4bac-9661-0ef5ee7732d3)
The output(PDU) is here:
[scenerio_zero_output.txt](https://github.com/user-attachments/files/19063995/scenerio_zero_output.txt)

**Comment by ethanlin355** [comment]
Same issues for using bronze ric from the tutorial

**Comment by Thecave3** [comment]
I see a `[INFO ] [E2AP] Received SETUP-RESPONSE-SUCCESS` on the output you shared

**Comment by ethanlin355** [comment]
Yes but on wireshark there is no evidence shows that connection is successful

**Comment by ethanlin355** [comment]
Can u give me an successful connection example?

**Comment by Thecave3** [comment]
> I see a `[INFO ] [E2AP] Received SETUP-RESPONSE-SUCCESS` on the output you shared

This means that the SCTP socket is connected and that the E2AP Setup procedure worked as intended. Did you enable the indication messages report in your application?

**Comment by ethanlin355** [comment]
I think I just run the default command on ur tutorial: ./waf --run scratch/scenario-zero.cc

**Comment by ethanlin355** [comment]
Yes the sctp process is successful and do u know where can I see the gnb ids?

**Comment by Thecave3** [comment]
As reported in [the tutorial](https://openrangym.com/tutorials/ns-o-ran), you can see the gnb id from the E2 Term of the RIC and from the xApp


![](https://openrangym.com/assets/post-assets/ns-o-ran-terminals.png)

**Comment by ethanlin355** [comment]
Connections failure is because of of cpu resources is not enough,I want to run it in another machine but I need to change the port of e2sim I found `e2sim_def.h` file in e2sim/src/DEF change the port to 32222 and then I rebuild the source (`./build_e2sim.sh`) I tried to do scenario zero after that but the port number is still 36422

**Comment by Thecave3** [comment]
Port is passed through this parameter:

https://github.com/wineslab/ns-o-ran-ns3-mmwave/blob/8c42089d0859396589f0dbb5a4dca29327ba756c/src/mmwave/helper/mmwave-helper.cc#L298-L302
